snapshot/policy/scheduling_policy.go

```diff
@@ -45,14 +46,18 @@ func (t TimeOfDay) String() string {
 
 // SortAndDedupeTimesOfDay sorts the slice of times of day and removes duplicates.
 func SortAndDedupeTimesOfDay(tod []TimeOfDay) []TimeOfDay {
-	sort.Slice(tod, func(i, j int) bool {
-		if a, b := tod[i].Hour, tod[j].Hour; a != b {
-			return a < b
+	slices.SortFunc(tod, func(a, b TimeOfDay) int {
```

Comment:
This is great 🎉 

@lupusA Can you add a simple ("table-based") unit test for `SortAndDedupeTimesOfDay` please?

Reply:
Hey @julio-lopez 
will provide the test in the next days

Reply:
Test in #3535
The test passes with this changes.